### ISSUE TITLE
CRISP: Micro Epsilon Confocal IDT (IOC and OPI)

### ISSUE THREAD

**Issue by KathrynBaker** [ISISComputingGroup/IBEX#7502]
As a commercial user on CRISP I want to use the range finder to be certain of where my sample is.

## Acceptance Criteria
- [x] There is an emulator for the IDT as it is used (see Notes)
- [x] There are tests for the IOC
- [x] There is an IOC which reads the values as per the specifications in the notes
- [x] There is an OPI to display the two sensor values of interest
- [x] A task is created to reserve the IP Address, or the IP Address is reserved

## Notes
- There is already an IOC that may well already do this, it will need to be checked.
- There is a VI, this has already been reviewed by @KathrynBaker, and the details are below for what the IOC needs to do. The OPI will look different, and the handling of the TCP/IP side of things should use our standard wherever possible
- No search has been made yet for manuals as the existing LabVIEW code is straightforward
- There is a TCP connection, the IP Address should be a macro, the port should default to 1024 if it is a macro
- At present 188 bytes are read from the TCP port, there is no obvious reason for this magic number, but it works, so consider whether or not this is a magic number, a macro, or not needed under IBEX
- When the first four bytes of the string returned from the TCP port is "DATA":
	- The string after the word "DATA" is unflattened to a cluster of 8 unsigned 32 bit ints, with a little-endian byte-order
	- Cluster element 7 out of 8 is divided by 1 million (1000000) to get the value for sensor 1
	- Cluster element 8 of 8 is divided by 1 million (1000000) to get the value for sensor 2
	- No other cluster elements are used

**Comment by NikolaRoev** [comment]
Check https://github.com/ISISComputingGroup/IBEX/issues/5365 to see what needs to be done for this issue.

**Comment by KathrynBaker** [comment]
On talking to GL for reflectometers this was mentioned as something to put on hold as it may not be needed that soon. I've left it here as this ticket has at least been looked at briefly so that we discuss it.

**Comment by Tom-Willemsen** [comment]
I've reviewed the existing IOC against the acceptance criteria above - all functionality mentioned above (including emulator/tests/opi) is already covered by the existing `MECRF` ioc.

For the avoidance of doubt, the relevant manual is at `\\isis\shares\ISIS_Experiment_Controls\Manuals\Micro-Epsilon Confocal rangefinder` and the wiki page for the device is at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Micro-Epsilon-confocal-rangefinder

I've added a tasks to the task board to reserve the IP address of this device as and when appropriate.

To review: verify that I've added the task mentioned above to the tasks board, and that the manuals / wiki page / ioc exist. The IOC itself has already been reviewed as part of a previous ticket, so no need to review the code.
